### PR TITLE
Adds a viscosity scaled pressure mass matrix solver.

### PR DESCRIPTION
**Pull request by alistairbntl**
This is the first of several PR that will begin integrating new preconditioning methods in Proteus for Stokes and Navier-Stokes solves.

### CODE REVIEW

File: proteus/LinearSolvers.py [hunk 1]
```diff
@@ -577,10 +516,496 @@ def failed(self):
...
+        """
+        Qsys_petsc4py = self._massMatrix(recalculate = recalculate)
+        self.Qp = Qsys_petsc4py.getSubMatrix(self.linear_smoother.isp,
+                                             self.linear_smoother.isp)
+        if output_matrix is True:
+            self._exportMatrix(self.Qp,"Qp")
+        return self.Qp
+
+    def _massMatrix(self,recalculate=False):
+        """ Generates a the mass matrix.
```
Comment (cekees) [inline proteus/LinearSolvers.py]: Could fix comment

File: proteus/LinearSolvers.py [hunk 2]
```diff
@@ -577,10 +516,496 @@ def failed(self):
...
+                                                                                              ksp.atol,
+                                                                                              ksp.rtol))
+            if truenorm < self.rnorm0*ksp.rtol:
+                return p4pyPETSc.KSP.ConvergedReason.CONVERGED_RTOL
+            if truenorm < ksp.atol:
+                return p4pyPETSc.KSP.ConvergedReason.CONVERGED_ATOL
+        return False
+
+    def _setConstantPressureNullSpace(self,global_ksp):
+        import pdb ; pdb.set_trace()
```
Comment (cekees) [inline proteus/LinearSolvers.py]: remove debugging

File: proteus/NonlinearSolvers.py [hunk 3]
```diff
@@ -10,7 +10,7 @@
 from math import *
 import math #to disambiguate math.log and log
 from LinearAlgebraTools import *
-from LinearSolvers import *
+#from LinearSolvers import *
```
Comment (cekees) [inline proteus/NonlinearSolvers.py]: delete line?

File: proteus/NumericalSolution.py [hunk 4]
```diff
@@ -255,6 +255,7 @@ def __init__(self,so,pList,nList,sList,opts,simFlagsList=None):
                     mlMesh = MeshTools.MultilevelTriangularMesh(0,0,0,skipInit=True,
                                                                 nLayersOfOverlap=n.nLayersOfOverlapForParallel,
                                                                 parallelPartitioningType=n.parallelPartitioningType)
+                    logEvent("NAHeader GridRefinements %i" % (n.nLevels) )
```
Comment (cekees) [inline proteus/NumericalSolution.py]: What is NAHeader?

Comment (cekees) [inline proteus/NumericalSolution.py]: Is that a label you use to extra data from the log?

Comment (alistairbntl) [inline proteus/NumericalSolution.py]: Yes, NAHeader is one of the text flags I use to extract data for the residual analysis.  That said, I don't want to pollute the output unnecessarily for other users.  Perhaps I should set the log-levels for these rather high or create a separate profiling method entirely?

**Pull request merged**